File: Unidad 1/T1.1 SIN BD/TI1.1 SOAP_JAVA_CONUNI_GR01/02.CLIESC/src/ec/edu/monster/service/ConversionService.java
package ec.edu.monster.service;

public class ConversionService {

    // ====== LOGIN ======
    public String login(String usuario, String password) {
        return WSClientFactory.getPort().login(usuario, password); // "SUCCESS" / "FAIL"
    }

    // ====== LONGITUD ======
    public double kilometrosAmillas(double km)     { return WSClientFactory.getPort().kilometrosAmillas(km); }
    public double metrosApies(double m)            { return WSClientFactory.getPort().metrosApies(m); }
    public double pulgadasAcentimetros(double in)  { return WSClientFactory.getPort().pulgadasAcentimetros(in); }

    // ====== MASA ======
    public double kilogramosAlibras(double kg)     { return WSClientFactory.getPort().kilogramosAlibras(kg); }
    public double gramosAonzas(double g)           { return WSClientFactory.getPort().gramosAonzas(g); }
    public double librasAkilogramos(double lb)     { return WSClientFactory.getPort().librasAkilogramos(lb); }

    // ====== TEMPERATURA ======
    public double celsiusAfahrenheit(double c)     { return WSClientFactory.getPort().celsiusAfahrenheit(c); }
    public double fahrenheitAcelsius(double f)     { return WSClientFactory.getPort().fahrenheitAcelsius(f); }
    public double celsiusAkelvin(double c)         { return WSClientFactory.getPort().celsiusAkelvin(c); }
}

File: Unidad 1/T1.1 SIN BD/TI1.1 SOAP_DOTNET_CONUNI_GR01/01 SERVIDOR/WS_SOAP_DOTNET_GRO01/App_Data/ConversionService.cs
using System;
using WS_SOAP_DOTNET_GR001.Services;

namespace WS_SOAP_DOTNET_GR001.Services
{
    public class ConversionService : IConversionService
    {
        // Longitud
        public double KilometrosAMillas(double km) => km * 0.621371;
        public double MetrosAPies(double m) => m * 3.28084;
        public double PulgadasACentimetros(double p) => p * 2.54;

        // Masa
        public double KilogramosALibras(double kg) => kg * 2.20462;
        public double GramosAOnzas(double g) => g * 0.035274;
        public double LibrasAKilogramos(double lb) => lb * 0.45359237;

        // Temperatura
        public double CelsiusAFahrenheit(double c) => (c * 9.0 / 5.0) + 32.0;
        public double FahrenheitACelsius(double f) => (f - 32.0) * 5.0 / 9.0;
        public double CelsiusAKelvin(double c) => c + 273.15;
    }
}

